latch_cli/docker_utils/__init__.py

```diff
@@ -0,0 +1,197 @@
+
+import yaml
+
+from latch_cli.types import LatchWorkflowConfig
+
+print = functools.partial(print, flush=True)
+
+
+class DockerCmdBlockOrder(Enum):
+    FIRST = auto()
+    PRECOPY = auto()
+    POSTCOPY = auto()
+    LAST = auto()
+
+
+@dataclass
+class DockerCmdBlock:
+    comment: str
+    commands: List[str]
+    order: DockerCmdBlockOrder
+
+
+def get_prologue(config: LatchWorkflowConfig) -> List[str]:
+    return [
+        "# latch base image + dependencies for latch SDK -- removing these will break the workflow",
+        f"from {config.base_image}",
+        f"run python3 -m pip install latch=={config.latch_version}",
+    ]
+
+
+def get_epilogue() -> List[str]:
+    return [
+        "# latch internal tagging system + expected root directory -- changing these lines will break the workflow",
+        "arg tag",
+        "env FLYTE_INTERNAL_IMAGE $tag",
+        "WORKDIR /root",
+    ]
+
+
+def infer_commands(pkg_root: Path) -> List[DockerCmdBlock]:
+    commands: List[DockerCmdBlock] = []
+
+    # apt requirements in workflow directory
+    if (pkg_root / "requirements.apt").exists():
+        print("Install apt requirements from `requirements.apt`")
+        commands.append(
+            DockerCmdBlock(
+                comment="install apt requirements",
+                commands=[
+                    "copy requirements.apt /root/requirements.apt",
+                    "run apt-get update -y && xargs apt-get install -y </root/requirements.apt",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            )
+        )
+
+    # R requirements in workflow directory
+    if (pkg_root / "environment.R").exists():
+        print("Install R packages from `environment.R`y")
+        commands.append(
+            DockerCmdBlock(
+                comment="install R requirements",
+                commands=[
+                    dedent(
+                        """\
+                        run apt-get update -y && \\
+                            apt-get install -y software-properties-common && \\
+                            add-apt-repository "deb http://cloud.r-project.org/bin/linux/debian buster-cran40/" && \\
+                            apt-get install -y r-base r-base-dev libxml2-dev libcurl4-openssl-dev libssl-dev wget \
+                        """
+                    ).strip(),
+                    "copy environment.R /root/environment.R",
+                    "run Rscript /root/environment.R",
+                ],
+                order=DockerCmdBlockOrder.PRECOPY,
+            )
+        )
+
+    # conda environment in workflow directory
+    if (pkg_root / "environment.yml").exists():
+        print("Create conda environment from `environment.yml`")
+        with open(pkg_root / "environment.yml") as f:
+            conda_env = yaml.safe_load(f.read())
+            if "name" in conda_env:
+                env_name = conda_env["name"]
+            else:
+                env_name = "unnamed"
```

Comment:
would prefer something like `latch_env` or `default_env`

Reply:
idk it is neither -- it is clear here that the author forgot the name field in their environment.yml